### ISSUE TITLE
Naming

### ISSUE THREAD

**Issue by n8mills** [elihorne/smartboards#6]
would love the option to add custom name at the end.

e.g. A01_Introduction

i have many sections of an app that i would like to split into sections

**Comment by elihorne** [comment]
hey @n8mills - I can't think of a good way to allow you to add the custom name through my plugin, but I could update my naming mechanism such that if there was already a name (like "Introduction"), it would only put the A01 at the beginning instead of replacing the whole thing.

Would that do the trick for you?

**Comment by n8mills** [comment]
That would be great!

Then I can break it into sections!

Perfect :)


--- 

Nathan Mills
Human Factors Designer

Centre for Global eHeallth Innovation
Healthcare Human Factors
Toronto General Hospital

www.humanfactors.ca <http://www.humanfactors.ca/>
+1 (647) 209 6455

From:  Eli Horne <notifications@github.com>
Reply-To:  elihorne/smartboards
<reply+00d8779e94ee249a2374e1ac56bbf8e6f28e42c1588091ae92cf00000001123fa0739
2a169ce06491da7@reply.github.com>
Date:  Wednesday, October 21, 2015 at 2:50 PM
To:  elihorne/smartboards <smartboards@noreply.github.com>
Cc:  Nathan Mills <nathan@humanfactors.ca>
Subject:  Re: [smartboards] Naming (#6)


hey @n8mills <https://github.com/n8mills>  - I can't think of a good way to
allow you to add the custom name through my plugin, but I could update my
naming mechanism such that if there was already a name (like
"Introduction"), it would only put the A01 at the beginning instead of
replacing the whole thing.

Would that do the trick for you?


Reply to this email directly or view it on GitHub
<https://github.com/elihorne/smartboards/issues/6#issuecomment-149991520> .

**Comment by elihorne** [comment]
Hey @n8mills - I just pushed an update that will try to preserve artboard names.  Take it for a spin.  You can clear out the whole artboard name, give it your custom name, and then run "align and rename".  It will prepend with the row letter and column number, and then put your custom name after.   You can edit it at will.

**Comment by n8mills** [comment]
Thank you

Is there an option to insert a custom name name first > so all artboards
will be renamed then the standard A00.

I have 100¹s of artboards and renaming them manually will take some time.

For example the social/community area of the app is very large 50 odd
screens. To be able to select the artboards run the plugin with option to
add custom name and then your standard algin and rename.

Would be of great help and something all of my team would love!

Thanks for all your hard work mate!
--- 

Nathan Mills
Human Factors Designer

Centre for Global eHeallth Innovation
Healthcare Human Factors
Toronto General Hospital

www.humanfactors.ca <http://www.humanfactors.ca/>
+1 (647) 209 6455

From:  Eli Horne <notifications@github.com>
Reply-To:  elihorne/smartboards
<reply+00d8779e32bff41154eb3808590e0ca1fab6d11eee48df9292cf00000001124fa0389
2a169ce06491da7@reply.github.com>
Date:  Monday, November 2, 2015 at 6:05 PM
To:  elihorne/smartboards <smartboards@noreply.github.com>
Cc:  Nathan Mills <nathan@humanfactors.ca>
Subject:  Re: [smartboards] Naming (#6)


Hey @n8mills <https://github.com/n8mills>  - I just pushed an update that
will try to preserve artboard names.  Take it for a spin.  You can clear out
the whole artboard name, give it your custom name, and then run "align and
rename".  It will prepend with the row letter and column number, and then
put your custom name after.   You can edit it at will.


Reply to this email directly or view it on GitHub
<https://github.com/elihorne/smartboards/issues/6#issuecomment-153171868> .

**Comment by elihorne** [comment]
@n8mills I'm not sure I understand. If you run the plugin, it will keep any existing artboard names, and prepend with the row/column.  Are you saying that you want to bulk-apply a name to a range of artboards?

**Comment by n8mills** [comment]
Sorry for the confusion

Yes bulk renaming. So I can select all artboards and have the option to
rename them to ³community² it then will do the standard realign and rename
but it will have the custom option either at the start or end

E.g. ³COMMUNITY_A00² or ³A00_COMMUNITY²


Because I upload these boards to Zeplin and InVision and can¹t have boards
with he same names otherwise it will override the existing screen.

I sketch I have different pages fir each section of the mobile app > splash,
introduction screens, on boarding screens, how page, community, goals, etc
etc and if each are named ³A00² etc it doesn¹t work.


Like having 2 files with the same name in the same folder > the computer
won¹t allow this. Same for all these programs.
--- 

Nathan Mills
Human Factors Designer

Centre for Global eHeallth Innovation
Healthcare Human Factors
Toronto General Hospital

www.humanfactors.ca <http://www.humanfactors.ca/>
+1 (647) 209 6455

From:  Eli Horne <notifications@github.com>
Reply-To:  elihorne/smartboards
<reply+00d8779ee5b225418ea25f9c99c7067ed75db06ca092e0f392cf0000000112525e329
2a169ce06491da7@reply.github.com>
Date:  Wednesday, November 4, 2015 at 7:00 PM
To:  elihorne/smartboards <smartboards@noreply.github.com>
Cc:  Nathan Mills <nathan@humanfactors.ca>
Subject:  Re: [smartboards] Naming (#6)


@n8mills <https://github.com/n8mills>  I'm not sure I understand. If you run
the plugin, it will keep any existing artboard names, and prepend with the
row/column.  Are you saying that you want to bulk-apply a name to a range of
artboards?


Reply to this email directly or view it on GitHub
<https://github.com/elihorne/smartboards/issues/6#issuecomment-153909381> .

**Comment by elihorne** [comment]
@n8mills  Putting the numbers at the end is out of scope currently as the feature exists to keep the exported artboards in alphabetical order, and I don't want to add complexity to the options.

Bulk renaming is interesting as I can't find another plugin that does that

**Comment by n8mills** [comment]
Numbers at the start works then a custom name for the end would be great!

What happens after a user goes beyond Z?




--- 

Nathan Mills
Human Factors Designer

Centre for Global eHeallth Innovation
Healthcare Human Factors
Toronto General Hospital

www.humanfactors.ca <http://www.humanfactors.ca/>
+1 (647) 209 6455

From:  Eli Horne <notifications@github.com>
Reply-To:  elihorne/smartboards
<reply+00d8779e12978574eb84ba6838c8b9911afdbaa7f4a3d4e992cf0000000112548e469
2a169ce06491da7@reply.github.com>
Date:  Friday, November 6, 2015 at 10:50 AM
To:  elihorne/smartboards <smartboards@noreply.github.com>
Cc:  Nathan Mills <nathan@humanfactors.ca>
Subject:  Re: [smartboards] Naming (#6)


@n8mills <https://github.com/n8mills>   Putting the numbers at the end is
out of scope currently as the feature exists to keep the exported artboards
in alphabetical order, and I don't want to add complexity to the options.

Bulk renaming is interesting as I can't find another plugin that does that


Reply to this email directly or view it on GitHub
<https://github.com/elihorne/smartboards/issues/6#issuecomment-154445634> .

**Comment by elihorne** [comment]
I just pushed an update to include a "rename page artboards" (cmd + opt + r) function. It will not clear out the A00_ stuff if its already there, but it will rename all artboards on that current page.

I also updated the numbering so that it supports two characters at the front (A00 - Z00 followed by AA00 - AZ00).  I may have to change this in the future, but it seems like a bit of an edge case for now. 

Take a look and let me know if its working for you.

**Comment by n8mills** [comment]
Thanks eli

The AZ will be helpful for my team for large apps that our company works on
this is an every day issue.

I¹ll download and take a look!
--- 

Nathan Mills
Human Factors Designer

Centre for Global eHeallth Innovation
Healthcare Human Factors
Toronto General Hospital

www.humanfactors.ca <http://www.humanfactors.ca/>
+1 (647) 209 6455

From:  Eli Horne <notifications@github.com>
Reply-To:  elihorne/smartboards
<reply+00d8779e12eb5b40707cd5f8fdc51f8acbfc1eb4d82f13e392cf00000001125615049
2a169ce06491da7@reply.github.com>
Date:  Saturday, November 7, 2015 at 2:37 PM
To:  elihorne/smartboards <smartboards@noreply.github.com>
Cc:  Nathan Mills <nathan@humanfactors.ca>
Subject:  Re: [smartboards] Naming (#6)


I just pushed an update to include a "rename page artboards" (cmd + opt + r)
function. It will not clear out the A00_ stuff if its already there, but it
will rename all artboards on that current page.

I also updated the numbering so that it supports two characters at the front
(A00 - Z00 followed by AA00 - AZ00).  I may have to change this in the
future, but it seems like a bit of an edge case for now.

Take a look and let me know if its working for you.


Reply to this email directly or view it on GitHub
<https://github.com/elihorne/smartboards/issues/6#issuecomment-154741937> .

**Comment by n8mills** [comment]
Just updated and tested 1.2 > its great! Thank you!
--- 

Nathan Mills
Human Factors Designer

Centre for Global eHeallth Innovation
Healthcare Human Factors
Toronto General Hospital

www.humanfactors.ca <http://www.humanfactors.ca/>
+1 (647) 209 6455

From:  Eli Horne <notifications@github.com>
Reply-To:  elihorne/smartboards
<reply+00d8779e12eb5b40707cd5f8fdc51f8acbfc1eb4d82f13e392cf00000001125615049
2a169ce06491da7@reply.github.com>
Date:  Saturday, November 7, 2015 at 2:37 PM
To:  elihorne/smartboards <smartboards@noreply.github.com>
Cc:  Nathan Mills <nathan@humanfactors.ca>
Subject:  Re: [smartboards] Naming (#6)


I just pushed an update to include a "rename page artboards" (cmd + opt + r)
function. It will not clear out the A00_ stuff if its already there, but it
will rename all artboards on that current page.

I also updated the numbering so that it supports two characters at the front
(A00 - Z00 followed by AA00 - AZ00).  I may have to change this in the
future, but it seems like a bit of an edge case for now.

Take a look and let me know if its working for you.


Reply to this email directly or view it on GitHub
<https://github.com/elihorne/smartboards/issues/6#issuecomment-154741937> .
